python/pyspark/sql/streaming/stateful_processor.py

```diff
@@ -274,8 +274,12 @@ def getValueState(
             resets the expiration time to current processing time plus ttlDuration.
             If ttl is not specified the state will never expire.
         """
-        self.stateful_processor_api_client.get_value_state(state_name, schema, ttl_duration_ms)
-        return ValueState(ValueStateClient(self.stateful_processor_api_client), state_name, schema)
+        schema_struct = self.stateful_processor_api_client.get_value_state(
```

Comment:
nit: It's not very straightforward to understand we request initialization for State and we get the schema as a result. (Arguably it's already confusing we get nothing from get_XXX, though it's following the current name convention.)

Why not just have a separate method? It's not very heavyweight even we have a separate request call(s) for schema, and it's only used for string schema.

Reply:
But I'm open for voices, I know this reduces one round trip and there would be people who prefers performance over cleaner code.
cc. @anishshri-db It'd be good to hear about your input.

Reply:
yea - probably better to have a separate API for this. As @HeartSaVioR mentioned, could we only do the conversion if string is passed ?

Reply:
Makes sense, will update, thanks!